bigbluebutton-html5/imports/ui/components/presentation/presentation-uploader/service.js

```diff
@@ -75,13 +77,51 @@ const observePresentationConversion = (meetingId, filename, onConversion) =>
     });
   });
 
-const uploadAndConvertPresentation = (file, meetingID, endpoint, onUpload, onProgress, onConversion) => {
+const requestPresentationUploadToken = (podId, meetingId, filename) =>
+  new Promise((resolve, reject) => {
+    makeCall('requestPresentationUploadToken', podId, filename);
+
+    let computation = null;
+    const timeout = setTimeout(() => {
+      computation.stop();
+      reject({ code: 408, message: 'requestPresentationUploadToken timeout' });
+    }, TOKEN_TIMEOUT);
+
+    Tracker.autorun((c) => {
+      computation = c;
+      const sub = Meteor.subscribe('presentation-upload-token', Auth.credentials, podId, filename);
+      if (!sub.ready()) return;
+
+      const PresentationToken = PresentationUploadToken.findOne({
+        podId,
+        meetingId,
+        filename,
+      });
+
+      if (!PresentationToken || !('failed' in PresentationToken)) return;
```

Comment:
Just documenting this a bit further. On token request fail you do not add `failed: boolean` in the presentation-upload-token collection. On token request pass you add `failed: false` as a default.
That said, here `!('failed' in PresentationToken)` means `if the request for token returned as 'success'`

Reply:
I missed that line. It would be more obvious if this setter was moved out like the "doc" you have in Pass handler